src/engine/FileDropHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   Handles signal from Godot when files are dragged and dropped onto the game window
+/// </summary>
+public class FileDropHandler : Node
+{
+    public override void _Ready()
+    {
+        GetTree().Connect("files_dropped", this, "OnFilesDropped");
+    }
+
+    private void OnFilesDropped(string[] files, int screen)
+    {
+        foreach (var file in files)
```

Comment:
Do we want to attempt to load multiple save files at a time? Is there any use case for supporting a multi-drop at this stage?

Reply:
I imagined that things like dropping individual creations (if we add export from the editor functionality) would be nice to load at the same time. And if https://github.com/Revolutionary-Games/Thrive/issues/2424 is done then it makes sense for saves. Though currently this crashes, but I think once https://github.com/Revolutionary-Games/Thrive/pull/2421 is merged this wouldn't crash anymore.

Reply:
Fair enough. I'm against this kind of future-proofing multiple PRs ahead, but that's a tactical choice you are entitled to make.

Reply:
I mean it would require *more* code to explicitly stop after the first save load has happened

Reply:
Nonsense: `HandleFileDrop(files[0], screen);`

Reply:
There needs to be separate logic to allow the save loading to act differently than say other files dropped. So that would break the overall design (for example adding another file type like an exported creature, wouldn't be good to just pass the first file to it). Only thing that would work is if a drop handler method could return a bool indicating that further files shouldn't be handled, or something (but with mixed file types dropped that wouldn't be the right thing).

Reply:
Well again, if you're future-proofing then taking my suggestion would be a bad idea. I was just saying that my tactic would have been to write a method that only works with what the code is expected to do today, then refactor to support more functionality when it is implemented.

Reply:
I always over futureproof my code. I don't have time right now but if I did, I would implement the extra features on top right now to end this discussion.